components/browser/state/src/main/java/mozilla/components/browser/state/state/NavigationState.kt

```diff
@@ -0,0 +1,16 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.state.state
+
+/**
+ * A value type holding navigation state information for a Session.
+ *
+ * @property canGoBack whether or not there's an history item to go back to.
+ * @property canGoForward whether or not there's an history item to go forward to.
+ */
+data class NavigationState(
+    val canGoBack: Boolean = false,
+    val canGoForward: Boolean = false
+)
```

Comment:
If we consider only dispatching the actual change (forward/back, see above)  then we could also consider flattening the state and adding those fields to `ContentState` directly?